app/agents/voice/breeze_buddy/workflows/order_confirmation/websocket_bot.py

```
@@ -773,60 +775,47 @@ async def _handle_address_incorrect(self):
-
-        # Remove empty parts and join with comma (no spaces for webhook format)
-        final_parts = [p for p in parts if p.strip()]
-        self.updated_address = ",".join(final_parts)
+    def _update_address_field(self, field_name: str, value, expected_key: str = None):
+        """Helper function to update address fields and reduce redundancy"""
+        # Handle dict objects directly (LLM passes dict objects, not strings)
+        if isinstance(value, dict):
+            if expected_key and expected_key in value:
+                clean_value = str(value[expected_key]).strip()
+            else:
+                clean_value = str(list(value.values())[0]).strip()
```

Ruff (0.13.3)

Prefer `next(iter(value.values()))` over single element slice

Replace with `next(iter(value.values()))`

(RUF015)